server/src/main/java/org/elasticsearch/cluster/ClusterInfo.java

```diff
@@ -255,6 +266,16 @@ public Map<String, DiskUsage> getNodeMostAvailableDiskUsages() {
         return this.mostAvailableSpaceUsage;
     }
 
+    /**
+     * Returns a map of shard IDs to the write-loads for use in balancing. The write-loads can be interpreted
+     * as the average number of threads that ingestion to the shard will consume.
+     * This information may be partial or missing altogether under some circumstances. The absence of a shard
+     * write load from the map should be interpreted as "unknown".
+     */
+    public Map<ShardId, Double> getShardWriteLoads() {
+        return shardWriteLoads;
+    }
+
```

Comment:
update the `#equals`, `#hashCode` methods below.

Reply:
I don't like that we add equals and hashCode just for testing, but I updated them in cd3169fabdc21c00193a89b9b6f3d06c4e7a0a4f